File: puzzles/ts/dont-panic1/dont-panic1.ts
let elevators = new Map();

const inputs: string[] = readline().split(" ");
const exitFloor: number = Number(inputs[3]); // floor on which the exit is found
const exitPos: number = Number(inputs[4]); // position of the exit on its floor
const nbElevators: number = Number(inputs[7]);
for (let i: number = 0; i < nbElevators; i++) {
  const inputs: string[] = readline().split(" ");
  const elevatorFloor: number = Number(inputs[0]);
  const elevatorPos: number = Number(inputs[1]);
  elevators.set(elevatorFloor, elevatorPos);
}
// For the last floor, we use the exit instead of an elevator.
elevators.set(exitFloor, exitPos);

// game loop
while (true) {
  const inputs: string[] = readline().split(" ");
  const cloneFloor: number = Number(inputs[0]);
  const clonePos: number = Number(inputs[1]);
  const direction: string = inputs[2];

  // determine action to take based on current state
  if (cloneFloor === -1) {
    console.log("WAIT");
  } else if (direction === "LEFT" && clonePos < elevators.get(cloneFloor)) {
    console.log("BLOCK");
  } else if (direction === "RIGHT" && clonePos > elevators.get(cloneFloor)) {
    console.log("BLOCK");
  } else {
    console.log("WAIT");
  }
}

File: puzzles/kotlin/src/dont-panic1/dont-panic1.kt
import java.util.*

fun main(args : Array<String>) {
  val input = Scanner(System.`in`)
  val nbFloors = input.nextInt() // number of floors
  val width = input.nextInt() // width of the area
  val nbRounds = input.nextInt() // maximum number of rounds
  val exitFloor = input.nextInt() // floor on which the exit is found
  val exitPos = input.nextInt() // position of the exit on its floor
  val nbTotalClones = input.nextInt() // number of generated clones
  val nbAdditionalElevators = input.nextInt() // ignore (always zero)
  val nbElevators = input.nextInt() // number of elevators
  var elevators = Array(nbElevators + 1, {0})
  // For the last floor, we use the exit instead of an elevator.
  elevators[exitFloor] = exitPos

  for (i in 0 until nbElevators) {
    val elevatorFloor = input.nextInt() // floor on which this elevator is found
    val elevatorPos = input.nextInt() // position of the elevator on its floor
    elevators[elevatorFloor] = elevatorPos
  }

  // game loop
  while (true) {
    val cloneFloor = input.nextInt() // floor of the leading clone
    val clonePos = input.nextInt() // position of the leading clone on its floor
    val direction = input.next() // direction of the leading clone: LEFT or RIGHT

    if (cloneFloor == -1) {
      println("WAIT")
    } else if (direction.equals("LEFT") && clonePos < elevators[cloneFloor]) {
      println("BLOCK")
    } else if (direction.equals("RIGHT") && clonePos > elevators[cloneFloor]) {
      println("BLOCK")
    } else {
      println("WAIT")
    }
  }
}
